llm/src/prompts/spec-and-schema-gen.ts

```diff
@@ -0,0 +1,224 @@
+[Concise specification that captures only the essential requirements]
+</spec>
+
+<plan>
+[Brief 3-step implementation plan using the existing schema]
+</plan>
+
+GUIDELINES:
+- Aim for the simplest possible solution that works with the existing schema
+- The specification should take into account the existing schema structure
+- Focus on what can be achieved quickly with the existing data model
+- Avoid suggesting modifications to the schema if possible
+
+IMPORTANT:
+- Focus on the simplest working version
+- The user can always iterate and improve the solution later
+`;
+
+/**
+ * Generates a complete specification, schema, and plan from a goal.
+ * @param goal The user's goal or request
+ * @param existingSchema Optional existing schema to use as a basis
+ * @returns Object containing title, description, specification, schema, and optional data
+ */
+export async function generateSpecAndSchema(
+  goal: string,
+  existingSchema?: JSONSchema,
+): Promise<{
+  title: string;
+  description: string;
+  spec: string;
+  schema: JSONSchema;
+  data?: any;
+}> {
+  let systemPrompt, userContent;
+
+  if (existingSchema) {
+    // When we have an existing schema, focus on generating specification
+    systemPrompt = SPEC_FROM_SCHEMA_PROMPT;
+    userContent = `
+Goal: ${goal}
+
+Existing Schema:
+\`\`\`json
+${JSON.stringify(existingSchema, null, 2)}
+\`\`\`
+
+Based on this goal and the existing schema, please provide a title, description, detailed specification, and implementation plan.
+`;
+  } else {
+    // When generating from scratch, use the full schema generation prompt
+    systemPrompt = SCHEMA_FROM_GOAL_PROMPT;
+    userContent = goal;
+  }
+
+  // Send the request to the LLM
+  const response = await client.sendRequest({
+    model: "anthropic:claude-3-7-sonnet-latest",
+    system: systemPrompt,
+    stream: false,
+    messages: [
+      {
+        role: "user",
+        content: userContent,
+      },
+    ],
+  });
+
+  // Extract sections from the response
+  const title = parseTagFromResponse(response, "title") || "New Charm";
+  const description = parseTagFromResponse(response, "description") ||
+    `A tool to ${goal}`;
+  const spec = parseTagFromResponse(response, "spec") || goal;
+  const plan = parseTagFromResponse(response, "plan");
+
+  // If we have an existing schema, use it; otherwise parse the generated schema
+  let schema: JSONSchemaWritable;
+  let data;
+
+  if (existingSchema) {
+    // Use the existing schema, no need to parse one
+    schema = { ...existingSchema };
+  } else {
+    // Parse the generated schema
+    const schemaJson = parseTagFromResponse(response, "schema");
+    const exampleData = parseTagFromResponse(response, "example_data");
+
+    try {
+      schema = schemaJson ? JSON.parse(schemaJson) : {};
+    } catch (error) {
+      console.error("Error parsing schema:", error);
+      // Fallback to an empty schema
+      schema = {};
+    }
+
+    // Parse example data if provided
+    try {
```

Comment:
we don't need example data, and definitely we shouldn't use it. that's what the defaults in the schema are for.

Reply:
removed